eclair-core/src/main/scala/fr/acinq/eclair/ShortChannelId.scala

```diff
@@ -56,6 +65,8 @@ object ShortChannelId {
   def outputIndex(shortChannelId: ShortChannelId): Int = (shortChannelId.id & 0xFFFF).toInt
 
   def coordinates(shortChannelId: ShortChannelId): TxCoordinates = TxCoordinates(blockHeight(shortChannelId), txIndex(shortChannelId), outputIndex(shortChannelId))
+
+  def generateLocalAlias(): LocalAlias = new ShortChannelId(System.nanoTime()) with LocalAlias // TODO: fixme (duplicate, etc.)
```

Comment:
⚠️  needs fixing, this is a dummy impl that leaks information on the creation date of the channel.

Since generating an alias only happens at channel creation, we can afford doing something "costly". For example this could be an `AtomicLong` that we initialize at startup by checking all currently assigned `localAlias`, and then we do an `addAndGet` with a random increment when we need to generate a new alias.

Reply:
We cannot afford a duplicate here though and the space isn't that big. We should expect a collision for 5 billion attempts on 64 bits and only 80k attempts for 32 bits right? https://en.wikipedia.org/wiki/Birthday_attack

Reply:
It's true, the space isn't that big, but I'm afraid there may be privacy issues with generating those incrementally...I'll check what other implementations do

Reply:
I did some maths to confirm this, let me know if my calculations look correct:

- scids are 64 bits long
- we want to reserve the range [0; 2^22] for real scids
- that leaves us with 2^42 available values for aliases 
- we'll consider that we want to have 250k channels using aliases (~2^18)
- the probability of a collision is then p(n) ~= 2^36 / 2^43 ~= 0.8%

This is indeed not great! But using a counter that we increment leaks information about the approximate creation time of the channel, which could let an attacker find the channel outpoint on-chain...so I believe that what we should do instead is:

- generate an alias randomly in the [2^22:2^64] range
- check our alias map to see if it's already assigned and re-roll if it is

It means we'll probably need to send a message to the router to obtain a new alias, the router is where we can guarantee uniqueness.